src/main/java/io/github/thebusybiscuit/slimefun4/utils/UnbreakingAlgorithm.java

```diff
@@ -0,0 +1,26 @@
+package io.github.thebusybiscuit.slimefun4.utils;
+
+
+import io.github.thebusybiscuit.slimefun4.core.attributes.DamageableItem;
+
+/**
+ * This a enum evaluating and indicating a {@link DamageableItem} 's chance to be damaged
+ * depending if it is a tool or an armor
+ *
+ * @author RobotHanzo
+ */
+public enum UnbreakingAlgorithm {
+    TOOLS,
+    ARMOR;
+    public boolean evaluate(int unbreakingLevel){
+        if (this == TOOLS) {
+            return !(Math.random() < (1.0 / (unbreakingLevel + 1)));
+        }
+
+        if (this == ARMOR) {
+            return !(Math.random() < 0.6 + (0.4 / (unbreakingLevel + 1)));
+        }
+
+        return false;
+    }
+}
```

Comment:
Reyling on if-statements here is not really ideal.
We can just pass the method as a lambda expression in the constructor like this:

```suggestion
public enum UnbreakingAlgorithm {

    ARMOR(lvl -> !(Math.random() < 0.6 + (0.4 / (unbreakingLevel + 1))),
    TOOLS(lvl -> !(Math.random() < (1.0 / (lvl + 1))));

    private final IntFunction<Boolean> function;
    
    UnbreakingAlgorithm(@Nonnull IntFunction<Boolean> function) {
        this.function = function;
    }

    public boolean evaluate(int unbreakingLevel) {
        return function.apply(unbreakingLevel);
    }
}
```